packages/pds/src/actor-store/index.ts

```diff
@@ -31,44 +32,60 @@ type ActorStoreResources = {
 }
 
 export class ActorStore {
-  dbCache: LRUCache<string, ActorDb>
-  keyCache: LRUCache<string, Keypair>
+  dbCache: TTLCache<string, ActorDb>
+  keyCache: TTLCache<string, Keypair>
   reservedKeyDir: string
 
   constructor(
     public cfg: ActorStoreConfig,
     public resources: ActorStoreResources,
   ) {
-    this.dbCache = new LRUCache<string, ActorDb>({
+    this.dbCache = new TTLCache<string, ActorDb>({
+      ttl: MINUTE,
       max: cfg.cacheSize,
+      updateAgeOnGet: true,
+      checkAgeOnGet: false,
       dispose: async (db) => {
         await db.close()
       },
-      fetchMethod: async (did, _staleValue, { signal }) => {
-        const { dbLocation } = await this.getLocation(did)
-        const exists = await fileExists(dbLocation)
-        if (!exists) {
-          throw new InvalidRequestError('Repo not found', 'NotFound')
-        }
-
-        // if fetch is aborted then another handler opened the db first
-        // so we can close this handle and return `undefined`
-        return signal.aborted
-          ? undefined
-          : getDb(dbLocation, cfg.disableWalAutoCheckpoint)
-      },
     })
-    this.keyCache = new LRUCache<string, Keypair>({
+    this.keyCache = new TTLCache<string, Keypair>({
+      ttl: MINUTE,
       max: cfg.cacheSize,
-      fetchMethod: async (did) => {
-        const { keyLocation } = await this.getLocation(did)
-        const privKey = await fs.readFile(keyLocation)
-        return crypto.Secp256k1Keypair.import(privKey)
-      },
+      updateAgeOnGet: true,
+      checkAgeOnGet: false,
     })
     this.reservedKeyDir = path.join(cfg.directory, 'reserved_keys')
   }
 
+  private async fetchCacheKey(did: string) {
+    const got = this.keyCache.get(did)
+    if (got) return got
+    const { keyLocation } = await this.getLocation(did)
+    const privKey = await fs.readFile(keyLocation)
+    const keypair = await crypto.Secp256k1Keypair.import(privKey)
+    this.keyCache.set(did, keypair)
+    return keypair
+  }
```

Comment:
Without some kind of coalescing or abort mechanism, it's may not be too difficult for multiple callers to get past the `if (got) return got` line in parallel and each end-up generating the cached item.  I think the cache handles the disposal on `set()`, so hopefully it at least wouldn't result in a leak 👍